openverse_catalog/dags/providers/provider_api_scripts/inaturalist.py

```diff
@@ -1,111 +1,161 @@
+from typing import Dict
 
+from airflow.providers.postgres.hooks.postgres import PostgresHook
+from common.constants import POSTGRES_CONN_ID
+from common.licenses import LicenseInfo, get_license_info
 from common.loader import provider_details as prov
-from common.storage.image import ImageStore
+from providers.provider_api_scripts.provider_data_ingester import ProviderDataIngester
 
 
 logging.basicConfig(
-    format='%(asctime)s - %(name)s - %(levelname)s:  %(message)s',
-    level=logging.INFO
+    format="%(asctime)s - %(name)s - %(levelname)s:  %(message)s", level=logging.INFO
 )
 logger = logging.getLogger(__name__)
 
 PROVIDER = prov.INATURALIST_DEFAULT_PROVIDER
-SCRIPT_DIR = '/usr/local/airflow/openverse_catalog/dags/providers/provider_csv_load_scripts/inaturalist/'
-
-# This is set up here as a single thread linear process, to be more consisten with the 
-# structure of the provider dag factory, but if I were to start from scratch with airflow, 
-# each of these would be a task, and the dag would be 00 >> [01, 02, 03, 04] >> 05.
-# In dbt, the schema would be created separately, and steps 1-5 would each be a "model"
-# and dbt would navigate the dependencies more or less automagically.
-LOAD_PROCESS = [
-    '00_create_schema.sql',
-    '01_photos.sql',
-    '02_observations.sql',
-    '03_taxa.sql',
-    '04_observers.sql',
-    '05_final_tsv.sql'
-]
-CONNECTION_ID = os.getenv("AIRFLOW_CONN_POSTGRES_OPENLEDGER_TESTING")
-
-
-def run_sql_file(file_name, file_path=SCRIPT_DIR, conn_id=CONNECTION_ID):
-    """
-    The process is really written in SQL so this script just enables logging 
-    and monitoring jobs, but this is the basic function to run the SQL files 
-    for each step.
-    """
-    logger.info(f"Running {file_name} using DB connection {conn_id}")
-    result = 'SQL failed. See log for details.'
-    try:
-        assert file_name[-4:]=='.sql'
-        assert os.path.exists(file_path + file_name)
-        db = psycopg2.connect(conn_id)
-        cursor = db.cursor()
-        query = dedent(open(file_path + file_name, 'r').read())
-        cursor.execute(query)
-        result = cursor.fetchall()
-        db.commit()
-        logger.info("Success!")
-    except Exception as e:
-        logger.warning(f"SQL step failed due to {e}")
-    return result
-
-
-def main():
-    """
-    This is really just looping through the SQL steps defined above, with some additional logging.
-    """
-    logger.info("Begin: Inaturalist script")
-    
-    for f in LOAD_PROCESS:
-        image_count = run_sql_file(f)
-        logger.info(f"Results: {str(image_count)}")
-    logger.info(f"Total images pulled: {image_count}")
-    logger.info('Terminated!')
-
-
-if __name__ == '__main__':
-    main()
+SCRIPT_DIR = (
+    "/usr/local/airflow/openverse_catalog/dags/providers/provider_csv_load_scripts/"
+    + "inaturalist/"
+)
+PG_TO_JSON_TEMPLATE = dedent(
+    open(os.path.join(SCRIPT_DIR, "05_export_dbpage_to_json_template.sql")).read()
+)
```

Comment:
I think the best way to manage this would be to set the file path up relative to the current file, that way we don't have to worry about the prefix. Additionally, we always want to [open a file in a context manager when using `open` so we don't have dangling file pointers](https://book.pythontips.com/en/latest/context_managers.html#context-managers). One way we can avoid having to do that altogether though is with [`pathlib`](https://docs.python.org/3/library/pathlib.html)! That also makes the path separator system agnostic.

```suggestion
SCRIPT_DIR = Path(__file__).parents[1] / "provider_csv_load_scripts/inaturalist"
PG_TO_JSON_TEMPLATE = dedent((SCRIPT_DIR / "05_export_dbpage_to_json_template.sql").read_text())
```

Reply:
Yes! Thank you so much! I had thought that `.read()` at the end would take care of reading the whole file and closing it, but I'm really glad to know that `.read_text()` is the way to do that.